modules/reactor/include/meshgenerators/PolygonMeshGeneratorBase.h

```diff
@@ -401,4 +448,59 @@ class PolygonMeshGeneratorBase : public MeshGenerator
+   * @param intervals radial interval numbers of the main regions of the blocks
+   * @param inner_boundary_layer_fractions radial fractions of the blocks to be defined as the
+   * boundary layers near the inner boundaries
+   * @param inner_boundary_layer_intervals radial interval numbers of the blocks to be defined as
+   * the boundary layers near the inner boundaries
+   * @param inner_boundary_layer_biases bias growth factors of the elements within the blocks to be
+   * defined as the boundary layers near the inner boundaries
+   * @param outer_boundary_layer_fractions radial fractions of the blocks to be defined as the
+   * boundary layers near the outer boundaries
+   * @param outer_boundary_layer_intervals radial interval numbers of the blocks to be defined as
+   * the boundary layers near the outer boundaries
+   * @param outer_boundary_layer_biases bias growth factors of the elements within the blocks to be
+   * defined as the boundary layers near the outer boundaries
+   * @return bias terms describing radial node positions of multiple blocks
+   */
+  std::vector<std::vector<Real>>
+  biasTermsCalculator(const std::vector<Real> radial_biases,
+                      const std::vector<unsigned int> intervals,
+                      const std::vector<Real> inner_boundary_layer_fractions,
+                      const std::vector<unsigned int> inner_boundary_layer_intervals,
+                      const std::vector<Real> inner_boundary_layer_biases,
+                      const std::vector<Real> outer_boundary_layer_fractions,
+                      const std::vector<unsigned int> outer_boundary_layer_intervals,
+                      const std::vector<Real> outer_boundary_layer_biases) const;
+
+  /**
+   * Creates bias terms for a single block.
+   * @param radial_bias bias growth factor of the elements within the main region of the block
+   * @param intervals radial interval number of the main region of the block
+   * @param inner_boundary_layer_fraction radial fraction of the block to be defined as the
+   * boundary layer near the inner boundary
+   * @param inner_boundary_layer_intervals radial interval number of the block to be defined as
+   * the boundary layer near the inner boundary
+   * @param inner_boundary_layer_bias bias growth factor of the elements within the block to be
+   * defined as the boundary layer near the inner boundary
+   * @param outer_boundary_layer_fraction radial fraction of the block to be defined as the
+   * boundary layer near the outer boundary
+   * @param outer_boundary_layer_intervals radial interval number of the block to be defined as
+   * the boundary layer near the outer boundary
+   * @param outer_boundary_layer_bias bias growth factor of the elements within the block to be
+   * defined as the boundary layer near the outer boundary
+   * @return bias terms describing radial node positions of a single block
```

Comment:
let's be more precise here. 
This is the fraction of the total radius for each node right?
This is cumulative too right? As in b[n] = b[n-1] + radial distance between node n and n-1

Reply:
Yes. will modify this.